post: Our Engineering Culture (#56)
loadsmart/blog: content/posts/2021-08-30-engineering-culture/index.md

---
title: Our Engineering Culture
author: Gustavo Barbosa
twitter: gustavocsb
layout: post
lang: en
path: /blog/our-engineering-culture
date: 2021-08-30
comments: true
---
Hello, everyone!

We are happy to announce that today we are making our [engineering culture](https://github.com/loadsmart/culture) **open source**.

> There you’ll find several articles describing how we operate as an Engineering organization. Content varies from hiring and onboarding to architecture guidelines and tech stack. This is something that we were already sharing with new members through onboarding sessions and with candidates we would reach out to, but the information, in general, was very scattered.

Reviewer: ```suggestion
There you’ll find several articles describing how we operate as an Engineering organization. Content varies from hiring and onboarding to architecture guidelines and tech stack. This is information that we are already sharing with new members through onboarding sessions and with candidates we reach out to, but this information is generally scattered.
```

When we decided to make this move, there were mainly two goals in mind:
1. Create a single, centralized place for our engineers to look at and remember our main practices (internal alignment)

> 2. Show to potential new candidates a bit of our day-to-day work so that, if there’s identification, they might want to join the mothership (external exposure)

Reviewer: Not sure what is meant by `if there’s identification` in this context

Author: That was indeed weird. I've changed it a bit. Thanks @alexantoniomoran

### VCS and GitHub

> Things that were perfectly working years ago might not be in the present date. Processes, policies, practices, culture. They all change with time. An interesting side-effect of having ours on top of a version control system is that we can see how it evolves historically. Healthy conversations on decisions that will adapt how we work will only move us forward in the direction we want our tech organization to be.

Reviewer: ```suggestion
Things that were working perfectly in previous years may not be working as well presently. Processes, policies, practices, culture- they all change with time. An interesting side-effect of having our culture outlined on a version controlled system is that we can see how it evolves over time. Healthy conversations on decisions that will adapt how we work will only move us forward in the direction we want our tech organization to go.
```

> Another advantage of building this in the open is the social aspect of GitHub and the transparency it creates. Contributions such as new content, fix of typos, and others, can be done through pull requests. If we change how we do things here, one can simply create an issue for discussion and also submit a PR with the updated version of it. 

Reviewer: ```suggestion
Another advantage of cataloguing our engineering culture out in the open is that we gain the social aspects of GitHub and the transparency it creates. Contributions such as adding new content, fixing typos, discussing ideas, and others can be done through pull requests. If we change how we do things, one can simply create an issue for discussion and submit a PR with our updated outlook on a topic. 
```
